src/Hosting/test/FunctionalTests/ShutdownTests.cs

```diff
@@ -0,0 +1,161 @@
+                var logger = loggerFactory.CreateLogger(testName);
+
+                var applicationPath = Path.Combine(TestPathUtilities.GetSolutionRootDirectory("Extensions"),
+                    "src", "Hosting", "test", "testassets", "Microsoft.Extensions.Hosting.TestApp");
+
+                var deploymentParameters = new DeploymentParameters(
+                    applicationPath,
+                    RuntimeFlavor.CoreClr,
+                    RuntimeArchitecture.x64)
+                {
+                    TargetFramework = Tfm.NetCoreApp30,
+                    ApplicationType = ApplicationType.Portable,
+                    PublishApplicationBeforeDeployment = true,
+                    StatusMessagesEnabled = false
+                };
+
+                deploymentParameters.EnvironmentVariables["DOTNET_STARTMECHANIC"] = shutdownMechanic;
+
+                using (var deployer = new SelfHostDeployer(deploymentParameters, loggerFactory))
+                {
+                    var result = await deployer.DeployAsync();
+
+                    var started = new ManualResetEventSlim();
+                    var completed = new ManualResetEventSlim();
+                    var output = string.Empty;
+                    deployer.HostProcess.OutputDataReceived += (sender, args) =>
+                    {
+                        if (!string.IsNullOrEmpty(args.Data) && args.Data.StartsWith(StartedMessage))
+                        {
+                            started.Set();
+                            output += args.Data.Substring(StartedMessage.Length) + '\n';
+                        }
+                        else
+                        {
+                            output += args.Data + '\n';
+                        }
+
+                        if (output.Contains(CompletionMessage))
+                        {
+                            completed.Set();
+                        }
+                    };
+
+                    started.Wait(50000);
+
+                    if (!started.IsSet)
+                    {
+                        throw new InvalidOperationException("Application did not start successfully");
+                    }
+
+                    SendShutdownSignal(deployer.HostProcess);
+
+                    completed.Wait(50000);
+
+                    WaitForExitOrKill(deployer.HostProcess);
+
+                    if (!started.IsSet)
+                    {
+                        throw new InvalidOperationException($"Application did not write the expected output. The received output is: {output}");
+                    }
+
+                    output = output.Trim('\n');
+
+                    Assert.Equal(CompletionMessage, output);
+                }
+            }
+        }
+
+        private void SendShutdownSignal(Process hostProcess)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                SendSIGINT(hostProcess.Id);
+            }
+            /*
```

Comment:
Hm?

Reply:
I don' think the test runs on Windows and OSX yet. It's linux only.

Reply:
Why not just remove the commented out code?

Reply:
easier to uncomment than to figure out what to call. Let's keep this, it's test code and I like the convenience.